Src/Newtonsoft.Json/Newtonsoft.Json.csproj

```
@@ -40,6 +40,8 @@
     Disable that warning since local builds do not strong-name sign the assembly, causing a mistmatch when comparing
     the previously shipped assemblies and current.-->
     <NoWarn Condition="'$(SignAssembly)' != 'true'">$(NoWarn);CP0003</NoWarn>
+    <IsTrimmable Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))">true</IsTrimmable>
+    <IsAotCompatible Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net7.0'))">true</IsAotCompatible>
```

Thinking about this a bit deeper - we really only want the analyzers turned on. We don't want the `IsTrimmable` assembly metadata attribute to be put into the assembly. That is going to break existing apps that work today - like Blazor WASM.

```suggestion
    <EnableSingleFileAnalyzer Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))">true</EnableSingleFileAnalyzer >
    <EnableTrimAnalyzer Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))">true</EnableTrimAnalyzer >
    <EnableAotAnalyzerCondition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net7.0'))">true</EnableAotAnalyzer>
```

---

> I would have expected trimming the serialization library not to make the problem much worse than it already is.

In a perfect world, this would be true. But I've seen crazier things. I would error on the side of not changing the behavior at all with this change.